### DISCUSSION TITLE
CASL subject helper for Prisma requires every field?

### DISCUSSION THREAD

**Discussion by dariusj18** [stalniy/casl#565] [Q&A]
When trying to use the `subject()` helper to test an ability against an object it seems that the object requires every possible field from the Prisma table's type to exist in the object.  My use case is testing a dto's fields against a query, ex. can a user submit a post of that type.  It seems a lot of possible work to build an object with every possible field, ex `{ id, createdAt, isDeleted, title, type }`, just because I want to test a single field.  

Am I doing this wrong, is it ok to use Partial<PrismaModel> when building the PrismaAbility type to attempt to fix this?

Maybe instead like this? `abilities.can('read', subject('Post', { type: 'foo' } as Post))`

**Comment by ahmadalfy** [comment]
I don't think it can work that way. What I did was creating functions that returns objects for test data with all the necessary fields.

**Comment by stalniy** [comment]
Sure, you need to provide all fields of your entities, otherwise you won't be able to change permissions without updating your checks around the code. 

The calling side `ability.can(read', subject('Post', post))` must not know about permission condition. So, you do everything correctly, and yes you need to pass the full object
